docs/language/learn-ql/ql-etudes/river-crossing.ql

```diff
@@ -56,20 +56,20 @@ class State extends string {
 
   /** Returns the state that is reached after ferrying a particular cargo item. */
   State ferry(Cargo cargo) {
-    cargo = "Nothing" and result = renderState(man.flip(), goat, cabbage, wolf)
+    cargo = "Nothing" and result = renderState(man.other(), goat, cabbage, wolf)
     or
-    cargo = "Goat" and result = renderState(man.flip(), goat.flip(), cabbage, wolf)
+    cargo = "Goat" and result = renderState(man.other(), goat.other(), cabbage, wolf)
     or
-    cargo = "Cabbage" and result = renderState(man.flip(), goat, cabbage.flip(), wolf)
+    cargo = "Cabbage" and result = renderState(man.other(), goat, cabbage.other(), wolf)
     or
-    cargo = "Wolf" and result = renderState(man.flip(), goat, cabbage, wolf.flip())
+    cargo = "Wolf" and result = renderState(man.other(), goat, cabbage, wolf.other())
   }
 
   /**
    * Holds if predator and prey are on the same shore and the man
-   * is on the other shore.
+   * is not present.
    */
-  predicate eats(Shore predator, Shore prey) { predator = prey and man = predator.flip() }
+  predicate eats(Shore predator, Shore prey) { predator = prey and man = predator.other() }
 
```

Comment:
If we change the description, perhaps we should also change the code? (say to `man != predator`)

Reply:
That makes sense - thank you!